### PR TITLE
[sg] fix environment overrides for bazel and docker commands

### PR DESCRIPTION
**Pull request by jamesmcnamara**
As title. We weren't actually merging the bazel and docker commands with the overwrites so this simply implements that.

## Test plan
Unit test and ran sg start locally with an overwrite file.

### CODE REVIEW

**Comment by rafax** [comment]
I tested this locally - when I remove sg.config.yaml CODY_GATEWAY_ANTHROPIC_ACCESS_TOKEN and set it in sg.config.override.yaml, Cody Gateway fails to start on main (due to the env var not being set), but it starts correctly with this PR

File: dev/sg/internal/run/bazel_command.go [hunk 1]
```diff
@@ -88,6 +89,18 @@ func (bc BazelCommand) GetExecCmd(ctx context.Context) (*exec.Cmd, error) {
 	return exec.CommandContext(ctx, "bash", "-c", fmt.Sprintf("%s\n%s", bc.Config.PreCmd, cmd)), nil
 }
 
+// Overrides the behavior of this command with other command.
```
Comment (jhchabran) [inline dev/sg/internal/run/bazel_command.go:92]:
```suggestion
// Merge overrides the behavior of this command with other command.
```

(go docs expects the inline docs to start with the function name, we don't enforce this though)

File: dev/sg/internal/sgconf/config.go [hunk 2]
```diff
@@ -150,56 +151,59 @@ type Config struct {
 	Tests             map[string]*run.Command       `yaml:"tests"`
 }
 
-// Merges merges the top-level entries of two Config objects, with the receiver
-// being modified.
-func (c *Config) Merge(other *Config) {
+// Merges merges the top-level entries of two Config objects, using the
```
Comment (jhchabran) [inline dev/sg/internal/sgconf/config.go:154]:
```suggestion
// Merge merges the top-level entries of two Config objects, using the
```

File: dev/sg/internal/sgconf/config.go [hunk 3]
```diff
@@ -221,3 +225,7 @@ func (c *Config) GetEnv(key string) string {
 		return os.Getenv(lookup)
 	})
 }
+
+func ptr[T any](expr T) *T {
```
Comment (jhchabran) [inline dev/sg/internal/sgconf/config.go:229]: nit: might want to use https://github.com/sourcegraph/sourcegraph/blob/main/lib/pointers/ptr.go

**Review by jhchabran — APPROVED** [review]
bunch of little nits, already approving so you can merge it without a subsequent review

**Pull request merged**